### DISCUSSION TITLE
Copy on Select functionality?

### DISCUSSION THREAD

**Discussion by ilar** [wezterm/wezterm#1033] [Q&A]
Hey all! Was wondering if there was a way to get wezterm to copy on mouse selection. It's a feature/option in a few other terminals I've used and while I'm a big fan of wezterm so far, I'm really missing that functionality.

Iv'e searched the docs and the closest thing I've found was CopyTo, but that seems to need a keybinding.

**Comment by bew** [comment]
This should already be working by default.

The mouse action is called `CompleteSelection`: https://wezfurlong.org/wezterm/config/lua/keyassignment/CompleteSelection.html
You can see the various mouse events bound to this in https://wezfurlong.org/wezterm/config/mouse.html

Also, on which OS are you?
- On Linux, the (default) copy destination for mouse selection is the primary selection (like most other X11 apps)
- On Windows, the (default) copy destination is the main clipboard
- On MacOS I don't know `¯\_(ツ)_/¯`

**Reply by wez** [reply]
on macOS the destination is the main clipboard (or pasteboard as it is sometimes known)

**Comment by ilar** [comment]
I'm on linux, ubuntu specifically, and this doesn't work by default for some reason. hm.

**Reply by bew** [reply]
Could you try to debug with xclip?
Select something, then run:
```sh
echo -n "Clipboard: " ; xclip -o -selection clipboard ; echo
echo -n "Primary selection: " ; xclip -o ; echo
```

**Reply by Rahuum** [reply]
```
Clipboard: echo -n "Clipboard: " ; xclip -o -selection clipboard ; echo
echo -n "Primary selection: " ; xclip -o ; echo
Primary selection: term.lu
```

after copying 'term.lu'

So it seems to be trying to seperate out my clipboard and my primary selection. How would I get it to go to my clipboard instead?

**Reply by bew** [reply]
To make it always go to the clipboard when you select something, you can redefine the mouse binding that ends a selection:
```lua
return {
  mouse_bindings = {
    {
      event={Up={streak=1, button="Left"}},
      mods="NONE",
      action=wezterm.action{CompleteSelectionOrOpenLinkAtMouseCursor="Clipboard"},
      -- NOTE: the default action is:
      -- action=wezterm.action{CompleteSelectionOrOpenLinkAtMouseCursor="PrimarySelection"},
    },
  },
}
```
Read more about mouse bindings at https://wezfurlong.org/wezterm/config/mouse.html

**Comment by aplatypus** [comment]
I can toss in two more cents worth because there IS something funny going on.  At first I thought the copy-on-select was not happening between two Wezterm windows.  However some fiddling about demonstrated that the select was working IF-AND-ONLY-IF I used Shift+Insert and Not when I used Ctrl+Shift+V (normal paste key) in the second Wezterm window.  

Both windows use the same `wexterm.lau` file.  

As remarked above this works out-of-the-box in the first window opened.  And things are different with the 2nd window instance.  Perhaps that will suggest a clue for this behaviour???

**Reply by bew** [reply]
please open a new bug report with detailed information about your environment / versions / etc.., this was an old discussion and the copy implementation had a few changes since then.
Also make sure to try with Wezterm nightly as your issue might have been fixed already
(the last stable is outdated at this point)
